cmd/aries-agent-rest/startcmd/start.go

```diff
@@ -735,59 +751,30 @@ func createStoreProviders(parameters *agentParameters) (storage.Provider, error)
 		sleep = 1 * time.Second
 	)
 
-	switch {
-	case strings.EqualFold(parameters.dbParam.dbType, databaseTypeMemOption):
-		return mem.NewProvider(), nil
-	case strings.EqualFold(parameters.dbParam.dbType, databaseTypeLevelDBOption):
-		return leveldb.NewProvider(parameters.dbParam.url), nil
-	case strings.EqualFold(parameters.dbParam.dbType, databaseTypeCouchDBOption):
-		var store storage.Provider
-
-		err := backoff.RetryNotify(
-			func() error {
-				var openErr error
-				store, openErr = couchdbstore.NewProvider(parameters.dbParam.url,
-					couchdbstore.WithDBPrefix(parameters.dbParam.prefix))
-
-				return openErr
-			},
-			backoff.WithMaxRetries(backoff.NewConstantBackOff(sleep), parameters.dbParam.timeout),
-			func(retryErr error, t time.Duration) {
-				logger.Warnf(
-					"failed to connect to storage, will sleep for %s before trying again : %s\n",
-					t, retryErr)
-			},
-		)
-		if err != nil {
-			return nil, fmt.Errorf("failed to connect to storage at %s : %w", parameters.dbParam.dbType, err)
-		}
+	provider, supported := supportedStorageProviders[parameters.dbParam.dbType]
+	if !supported {
+		return nil, fmt.Errorf("key database type not set to a valid type." +
+			" run start --help to see the available options")
+	}
 
-		return store, nil
-
-	case strings.EqualFold(parameters.dbParam.dbType, databaseTypeMYSQLDBOption):
-		var store storage.Provider
-
-		err := backoff.RetryNotify(
-			func() error {
-				var openErr error
-				store, openErr = mysql.NewProvider(parameters.dbParam.url,
-					mysql.WithDBPrefix(parameters.dbParam.prefix))
-				return openErr
-			},
-			backoff.WithMaxRetries(backoff.NewConstantBackOff(sleep), parameters.dbParam.timeout),
-			func(retryErr error, t time.Duration) {
-				logger.Warnf(
-					"failed to connect to storage, will sleep for %s before trying again : %s\n",
-					t, retryErr)
-			},
-		)
-		if err != nil {
-			return nil, fmt.Errorf("failed to connect to storage at %s : %w", parameters.dbParam.dbType, err)
-		}
+	var store storage.Provider
 
-		return store, nil
+	err := backoff.RetryNotify(
+		func() error {
+			var openErr error
+			store, openErr = provider(parameters.dbParam.url, parameters.dbParam.prefix)
+			return openErr
+		},
+		backoff.WithMaxRetries(backoff.NewConstantBackOff(sleep), parameters.dbParam.timeout),
```

Comment:
nit: It would be nice to use explicitly `time.Second` instead of `sleep`.

Reply:
done